py-polars/src/series/mod.rs

```diff
@@ -154,9 +154,9 @@ impl PySeries {
         }
     }
 
-    fn get_idx(&self, py: Python, idx: usize) -> PyResult<PyObject> {
+    fn get_idx(&self, py: Python, idx: usize, wrap_lists: bool) -> PyResult<PyObject> {
```

Comment:
`wrap_lists` is really the wrong way to think about it. `Series` is the core representation. We can convert it into a list at some cost.

Reply:
> Series is the core representation. We can convert it into a list at some cost.

At this layer I believe they are ChunkedArrays, no? You have to pay a cost to convert to PySeries (and then call into python to further wrap as Series) vs simply reading out the array as a list.

Reply:
But a Series is just a wrapper for the ChunkedArray. Wrapping is cheap. If we convert to a Python list, there is a conversion, e.g. `UInt8` has to become `Int64` etc.

Reply:
> Wrapping is cheap. If we convert to a Python list, there is a conversion

It's demonstrably more expensive to return the wrapped series, uint or otherwise:
```python
df = pl.DataFrame({
        "a": ["xxx", "yyy"],
        "b": [[1,2,3,4,5], [6,7,8,9,10]],
    },
    schema_overrides={"b": pl.List(pl.UInt8)},
)
```
```bash
# returning _list_ from item (wrap_lists=False)
%timeit df.item(1,"b")
530 ns ± 20.8 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

# returning _series_ from item (wrap_lists=True)
%timeit df.item(1,"b")
709 ns ± 9.62 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
> And this parameter does not work correctly for nested lists:

The element being returned in that example is a Struct, not a List, so that is expected - currently _only_ Lists get wrapped (hence the parameter name, which states what's actually happening, and this behaviour also helps indicate why I don't like the weird inconsistency we have going on here).

I'd happily remove the parameter completely and just have us consistently return py-native data here rather than our somewhat half-baked special-casing of List 😅

Reply:
Maybe that's indeed better, let my brain churn on this one moment :smile: